src/parser/statement.js

```diff
@@ -79,7 +84,11 @@ pp.parseStatement = function (declaration, topLevel) {
       return this.parseClass(node, true);
 
     case tt._if: return this.parseIfStatement(node);
-    case tt._return: return this.parseReturnStatement(node);
+
+    case tt._return:
+      if (this.hasPlugin("lightscript") && isExprContext) this.unexpected(null, "return is illegal inside LightScript expressions");
```

Comment:
Let's instead use a `this.state.inExpressionBlock` or something, and inside `this.parseReturnStatement` / `this.parseBreakContinueStatement` do the check. 

Then we won't need to pass around `isExprContext` like this or modify `this.parseStatement` itself at all.